### PR TITLE
reimplement paddle.nn.functional.sigmoid_focal_loss

### PR DESCRIPTION
**Pull request by FlyingQianMM**
### PR types
New features

### PR changes
OPs

### Describe
The paddle.nn.functional.sigmoid_focal_loss is implemented by alias of paddle.fluid.layers.sigmod_focal_loss before, now it is reimplement by composing the sigmoid and binary_cross_entropy_with_logits ops.

chinese pr: https://github.com/PaddlePaddle/FluidDoc/pull/2751
image
image
image

### CODE REVIEW

File: python/paddle/nn/functional/loss.py [hunk 1]
```diff
@@ -1142,3 +1141,166 @@ def cross_entropy(input,
...
+    If :attr:`reduction` set to ``'none'``, the operator will return the original loss `Out`.
+    If :attr:`reduction` set to ``'mean'``, the reduced mean loss is :math:`Out = MEAN(Out)`.
+    If :attr:`reduction` set to ``'sum'``, the reduced sum loss is :math:`Out = SUM(Out)`.
+
+    Note that the target ``label`` is 0 for the negative class and is 1 for the positive class.
+
+    Args:
+        logit (Tensor): The input prediction tensor. 2-D tensor with shape: [N, *],
+            N is batch_size, `*` means number of additional dimensions. The ``logit``
+            is usually the output of a convolution layer. Available dtype is float32, float64.
```
Comment (wawltor) [inline python/paddle/nn/functional/loss.py]: 是否支持多维的tensor的输入了？* 表示额外的维度，* 有可能是一维或者多维的信息

File: python/paddle/nn/functional/loss.py [hunk 2]
```diff
@@ -1142,3 +1141,166 @@ def cross_entropy(input,
...
+        label (Tensor): The target label tensor. 2-D tensor with the same shape as
+            ``logit``. The target label whose value should be numbers between 0 and 1.
+            Available dtype is float32, float64.
+        normalizer (Tensor, optional): The number normalizes the focal loss. It has to be
+            a 1-D Tensor whose shape is `[1, ]`. The data type is float32, float64.
+            For object detection task, it is the the number of positive samples.
+            If set to None, the focal loss will not be normalized.
+        alpha(int|float): Hyper-parameter to balance the positive and negative example,
```
Comment (wawltor) [inline python/paddle/nn/functional/loss.py]: 因为有默认值，因此是optional

File: python/paddle/nn/functional/loss.py [hunk 3]
```diff
@@ -1142,3 +1141,166 @@ def cross_entropy(input,
...
+            it should be between 0 and 1.  Default value is set to 0.25. 
+        gamma(int|float): Hyper-parameter to modulate the easy and hard examples.
```
Comment (wawltor) [inline python/paddle/nn/functional/loss.py]: 同上

File: python/paddle/nn/functional/loss.py [hunk 4]
```diff
@@ -1142,3 +1141,166 @@ def cross_entropy(input,
...
+    Returns:
+        Tensor, if :attr:`reduction` is ``'mean'`` or ``'sum'``, the out shape is :math:`[1]`, otherwise the shape is the same as ``logit``. The same dtype as ``logit`` tensor.
+
+    Examples:
+
+        .. code-block:: python
+
+            import paddle
+
+            paddle.disable_static()
```
Comment (wawltor) [inline python/paddle/nn/functional/loss.py]: 尝试一下是否可以去掉disable_static，现在默认打开的是动态图

File: python/paddle/nn/functional/loss.py (hunk 1 above)
Comment (FlyingQianMM) [reply python/paddle/nn/functional/loss.py]: 已经去掉2-D tensor的描述。谢谢～

File: python/paddle/nn/functional/loss.py (hunk 2 above)
Comment (FlyingQianMM) [reply python/paddle/nn/functional/loss.py]: 已加上optional。谢谢～

File: python/paddle/nn/functional/loss.py (hunk 3 above)
Comment (FlyingQianMM) [reply python/paddle/nn/functional/loss.py]: 已加上，谢谢～

File: python/paddle/nn/functional/loss.py (hunk 4 above)
Comment (FlyingQianMM) [reply python/paddle/nn/functional/loss.py]: 已经去掉，可以正常运行。谢谢～

**Review by wawltor — APPROVED** [review]
LGTM
